src/VirtoCommerce.ExperienceApiModule.DigitalCatalog/Middlewares/EnsureCategoryLoadingMiddleware.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using PipelineNet.Middleware;
+using VirtoCommerce.CatalogModule.Core.Services;
+using VirtoCommerce.XDigitalCatalog.Queries;
+
+namespace VirtoCommerce.XDigitalCatalog.Middlewares
+{
+    public class EnsureCategoryLoadingMiddleware : IAsyncMiddleware<SearchCategoryResponse>
+    {
+        private readonly ICategoryService _categoryService;
+
+        public EnsureCategoryLoadingMiddleware(ICategoryService categoryService)
+        {
+            _categoryService = categoryService;
+        }
+
+        public async Task Run(SearchCategoryResponse parameter, Func<SearchCategoryResponse, Task> next)
+        {
+            var itemsIds = parameter.Results
+                .Where(expProduct => expProduct.Category is null)
+                .Select(expProduct => expProduct.Key)
+                .Where(key => key != null)
+                .ToArray();
+
+            if (itemsIds.Any())
+            {
+                string responceGroup = null/*parameter.Query.GetResponseGroup()*/; // todo: responce group for category
```

Comment:
It should be another method and response group:
https://github.com/VirtoCommerce/vc-module-catalog/blob/bac3d033464b06eabecc3a9f8a5b1bcc799004f7/src/VirtoCommerce.CatalogModule.Core/Model/CategoryResponseGroup.cs#L6